### ISSUE TITLE
Whether it can run on ios platform

### ISSUE THREAD

**Issue by sherryruo** [fujiapple852/trippy#840]

**Comment by fujiapple852** [comment]
@sherryruo Trippy is not available for Android or iOS today, though I would like it to be.

I haven't spent any time researching what would be required to support these platforms, however one clear problem is the need for raw sockets for unprivileged users (see discussion in #101).  Another likely issue would be Tui (`ratatui` and `crossterm`) support for these platforms.  Support for CI build and test in GH is another consideration.

I would like to support Android and iOS but, realistically, it won't happen any time soon.  If others who want this support could upvote this issue it would help to gauge the demand out there.  Of course i'm always open to community contributions if there are iOS/Android experts out there who may be willing to help on this.

**Comment by isjarvi** [comment]
Supporting iOS and Android is meaningful as it allows mobile devices to troubleshoot network issues.

**Comment by immanuelfodor** [comment]
Mobile is very useful for network debug purposes, indeed, it would be great if [Termux](https://github.com/termux) would be supported with the unprivileged mode. So I don't mean a full native Android app but a Linux CLI packaged for Termux.

**Comment by JumperTheHero** [comment]
> Mobile is very useful for network debug purposes, indeed, it would be great if [Termux](https://github.com/termux) would be supported with the unprivileged mode. So I don't mean a full native Android app but a Linux CLI packaged for Termux.

I agree with immanuel. If we just had a Termux package, that would be good! MTR doesn't really work out of the box on Termux anyway...lol.

**Comment by fujiapple852** [comment]
I’m not familiar with Termux at all, however I can see that a patched version of MTR is [packaged](https://github.com/termux/termux-packages/tree/master/root-packages/mtr) for it (for rooted devices only) so in principle Trippy could be packaged as well. 

What this needs is someone familiar with the Termux ecosystem who is motivated to create and maintain Trippy.

Alternatively a [package request]( https://github.com/termux/termux-packages/issues) could be raised. 

Finally I see that cargo is already available for Termux and so perhaps a simple `cargo install trippy —locked` would be all that is required.

**Comment by CalebBrendel** [comment]
@fujiapple852 Just so you know I tried installing trippy via cargo. Still no dice as trippy being installed with cargo does not work on Termux...tis a shame. Android is semi built on Linux...I feel like this is possible and I almost want to learn programming to accomplish it. LMFAO.

Thanks for letting me know all of that info btw! :)

**Comment by fujiapple852** [comment]
Thank you for trying it out @CalebBrendel.  Can you give some details on what the issues were? I.e, did it fail to compile or fail to start or did it run but did not show any data?
